libcudacxx/include/nv/detail/__target_macros

```diff
@@ -448,21 +469,90 @@
 // Re-enable sm_101a support in nvcc.
 #  undef NV_HAS_FEATURE_SM_101a
 #  define NV_HAS_FEATURE_SM_101a __NV_HAS_FEATURE_SM_101a
-#  if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 1010) && defined(__CUDA_ARCH_FEAT_SM101_ALL))
+#  if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 1010) \
+    && (defined(__CUDA_ARCH_FEAT_SM101_ALL)             \
+        || (defined(__CUDA_ARCH_FAMILY_SPECIFIC__) && __CUDA_ARCH_FAMILY_SPECIFIC__ == 1010))
 #    define _NV_TARGET_BOOL___NV_HAS_FEATURE_SM_101a 1
 #  else
 #    define _NV_TARGET_BOOL___NV_HAS_FEATURE_SM_101a 0
 #  endif
 
+// Re-enable sm_103a support in nvcc.
+#  undef NV_HAS_FEATURE_SM_103a
+#  define NV_HAS_FEATURE_SM_103a __NV_HAS_FEATURE_SM_103a
+#  if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 1030) \
+    && (defined(__CUDA_ARCH_FEAT_SM103_ALL)             \
+        || (defined(__CUDA_ARCH_FAMILY_SPECIFIC__) && __CUDA_ARCH_FAMILY_SPECIFIC__ == 1030))
+#    define _NV_TARGET_BOOL___NV_HAS_FEATURE_SM_103a 1
+#  else
+#    define _NV_TARGET_BOOL___NV_HAS_FEATURE_SM_103a 0
+#  endif
+
 // Re-enable sm_120a support in nvcc.
 #  undef NV_HAS_FEATURE_SM_120a
 #  define NV_HAS_FEATURE_SM_120a __NV_HAS_FEATURE_SM_120a
-#  if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 1200) && defined(__CUDA_ARCH_FEAT_SM120_ALL))
+#  if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 1200) \
+    && (defined(__CUDA_ARCH_FEAT_SM120_ALL)             \
+        || (defined(__CUDA_ARCH_FAMILY_SPECIFIC__) && __CUDA_ARCH_FAMILY_SPECIFIC__ == 1200))
 #    define _NV_TARGET_BOOL___NV_HAS_FEATURE_SM_120a 1
 #  else
 #    define _NV_TARGET_BOOL___NV_HAS_FEATURE_SM_120a 0
 #  endif
 
+// Re-enable sm_121a support in nvcc.
+#  undef NV_HAS_FEATURE_SM_121a
+#  define NV_HAS_FEATURE_SM_121a __NV_HAS_FEATURE_SM_121a
+#  if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 1210) \
+    && (defined(__CUDA_ARCH_FEAT_SM121_ALL)             \
+        || (defined(__CUDA_ARCH_FAMILY_SPECIFIC__) && __CUDA_ARCH_FAMILY_SPECIFIC__ == 1210))
+#    define _NV_TARGET_BOOL___NV_HAS_FEATURE_SM_121a 1
+#  else
+#    define _NV_TARGET_BOOL___NV_HAS_FEATURE_SM_121a 0
+#  endif
+
+//----------------------------------------------------------------------------------------------------------------------
+// arch-specific SM versions
+
+// Re-enable sm_100f support in nvcc.
+#  undef NV_HAS_FEATURE_SM_100f
+#  define NV_HAS_FEATURE_SM_100f __NV_HAS_FEATURE_SM_100f
+#  if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ == 1000) && defined(__CUDA_ARCH_SPECIFIC__) \
```

Comment:
IIUC, `_NV_TARGET_BOOL___NV_HAS_FEATURE_SM_100f` is enabled when the code is compiled with `-arch sm_100a`, but not when the code is compiled with `-arch sm_100f`. Is this correct?

Remark: Should this line not also mention `__CUDA_ARCH_FAMILY_SPECIFIC__` to detect `-arch sm_100f`?

Question: Do we want to map one-to-one to the `nvcc`-defined macros or do we want to have the convenience feature for users where `-arch sm_100a` happens to define `_NV_TARGET_BOOL___NV_HAS_FEATURE_SM_100f`, as implemented above? From a maintenance point-of-view, not having this convenience feature would be easier. I can imagine however, that it would be surprising if some code that works on all of 100f does not compile when `-arch sm_100a` is specified.